Source/Modules/swigmain.cxx

```diff
@@ -147,11 +148,90 @@ void SWIG_merge_envopt(const char *env, int oargc, char *oargv[], int *nargc, ch
   for (int i = 1; (i < oargc) && (argc < arge); ++i, ++argc) {
     argv[argc] = oargv[i];
   }
+  argv[argc] = NULL;
 
   *nargc = argc;
   *nargv = argv;
 }
 
+static void insert_option(int *argc, char ***argv, int index, char const* start, char const *end) {
+  int new_argc = *argc;
+  char **new_argv = *argv;
+  size_t option_len = end - start;
+
+  // Preserve the NULL pointer at argv[argc]
+  new_argv = (char **)realloc(new_argv, (new_argc + 2) * sizeof(char *));
+  memmove(&new_argv[index + 1], &new_argv[index], sizeof(char *) * (new_argc + 1 - index));
+  new_argc++;
+
+  new_argv[index] = (char *)malloc(option_len + 1);
+  memcpy(new_argv[index], start, option_len);
+  new_argv[index][option_len] = '\0';
+
+  *argc = new_argc;
+  *argv = new_argv;
+}
+
+void SWIG_merge_options_files(int *argc, char ***argv) {
```

Comment:
This should be a static function without the `SWIG_` prefix. `SWIG_merge_envopt` was there before but is also wrong and ought to be corrected.